### PR TITLE
added possibility to ask for expanded invocations of testruns

### PR DESCRIPTION
**Pull request by hos-p**
Added support to list all test runs with all the invocations flattened

GET http://teamcity:8111/app/rest/testOccurrences?locator=build:(id:XXX),test:(id:XXX),expandInvocations:true

### CODE REVIEW

File: teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt [hunk 1]
```diff
@@ -186,6 +186,7 @@ interface TestRunsLocator {
     fun forTest(testId: TestId): TestRunsLocator
     fun forProject(projectId: ProjectId): TestRunsLocator
     fun withStatus(testStatus: TestStatus): TestRunsLocator
+    fun withExpandInvocations(expandInvocations: Boolean) : TestRunsLocator
```
Comment (chashnikov) [inline teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt:189]: What does this method mean? It isn't evident from its name. Could you please write KDoc for it?

Comment (hos-p) [reply teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt:193]: It add possibility to list all invocations of test in run.

Comment (chashnikov) [reply teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt:193]: Sorry, I still don't get it. What does it mean "return all invocations"? As far as I can see even without this flag all test invocations satisfying other filters are returned. I tried running different queries on our TeamCity instance and didn't notice a difference with or without this flag.

And if the value is `false` by default one we need to have a boolean flag in the method? There seems to be no sense to call `withExpandInvocations(false)`.

Comment (hos-p) [reply teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt]: I removed useless param. 

image

This is the only documentataion about this field in teamcity rest api documentation. My use case is that I retry failed tests and I want to retrieve all invocation of tests. This option was the only one which allows me to retrieve all results of particular test.

Comment (chashnikov) [reply teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt]: I asked TeamCity guys about this parameter. It indeed specifies how to describe results when some test was executed several times in same build. By default such runs are aggregated into a single value, `duration` property will be the sum of durations of individual runs, and `status` will be `SUCCESSFUL` if and only if all runs are successful. If `expandInvocations` is `true` individual runs of such tests are reported as separate entries. I think we should describe this in KDoc. Also I think we can invent a better name for it. What about `expandMultipleInvocations`?

Comment (hos-p) [reply teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt]: Sure, make sense to me.

**Comment by chashnikov** [comment]
Thank you, I've merged the change and described it in [CHANGELOG](https://github.com/JetBrains/teamcity-rest-client/commit/8d27cd7d17a0524b035817d2dabc0881ff346c27).

**Comment by chashnikov** [comment]
1.12.0 which includes this change is now published.

**Pull request merged**